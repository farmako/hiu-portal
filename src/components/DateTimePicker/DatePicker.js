import React from 'react';
import * as PropTypes from 'prop-types';
import { KeyboardDatePicker } from '@material-ui/pickers';

const DatePicker = ({
  disableFuture,
  minDate,
  selectedDate,
  handleDateChange,
}) => (
  <KeyboardDatePicker
    autoOk="true"
    variant="inline"
    margin="normal"
    disableFuture={disableFuture}
    id="date-picker-dialog"
    minDate={minDate}
    format="dd/MM/yyyy"
    value={selectedDate}
    onChange={handleDateChange}
    KeyboardButtonProps={{
      'aria-label': 'change date',
    }}
  />
);

DatePicker.propTypes = {
  minDate: PropTypes.instanceOf(Date),
  disableFuture: PropTypes.bool,
  selectedDate: PropTypes.instanceOf(Date).isRequired,
  handleDateChange: PropTypes.func.isRequired,
};

DatePicker.defaultProps = {
  minDate: new Date(1900, 1, 1),
  disableFuture: false,
};

export default DatePicker;
