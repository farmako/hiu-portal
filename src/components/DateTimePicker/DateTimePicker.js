import React from 'react';
import * as PropTypes from 'prop-types';
import { KeyboardDateTimePicker } from '@material-ui/pickers';

const DateTimePicker = ({
  disableFuture,
  minDate,
  maxDate,
  selectedDate,
  handleDateChange,
  disablePast,
}) => (
  <KeyboardDateTimePicker
    autoOk="true"
    variant="inline"
    margin="normal"
    disableFuture={disableFuture}
    id="date-picker-dialog"
    format="dd/MM/yyyy HH:mm"
    minDate={minDate}
    maxDate={maxDate}
    value={selectedDate}
    onChange={handleDateChange}
    disablePast={disablePast}
    KeyboardButtonProps={{
      'aria-label': 'change date',
    }}
  />
);

DateTimePicker.propTypes = {
  minDate: PropTypes.instanceOf(Date),
  maxDate: PropTypes.instanceOf(Date),
  disableFuture: PropTypes.bool,
  disablePast: PropTypes.bool,
  selectedDate: PropTypes.instanceOf(Date).isRequired,
  handleDateChange: PropTypes.func.isRequired,
};

DateTimePicker.defaultProps = {
  minDate: new Date(1900, 1, 1),
  maxDate: new Date(2100, 1, 1),
  disableFuture: false,
  disablePast: false,
};

export default DateTimePicker;
