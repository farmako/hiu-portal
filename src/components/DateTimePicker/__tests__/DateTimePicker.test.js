import React from 'react';
import { shallow } from 'enzyme';
import DateTimePicker from '../DateTimePicker';

describe('DateTimePicker', () => {
  const minDate = new Date('2019-01-01T11:01:58.135Z');
  const maxDate = new Date('2019-12-31T11:01:58.135Z');
  const selectedDate = new Date('2020-04-22T03:17:48.603+0000');
  const handleDateChange = jest.fn();
  const wrapper = shallow(
    <DateTimePicker
      minDate={minDate}
      maxDate={maxDate}
      disableFuture={false}
      selectedDate={selectedDate}
      disablePast={false}
      handleDateChange={handleDateChange}
    />,
  );

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
