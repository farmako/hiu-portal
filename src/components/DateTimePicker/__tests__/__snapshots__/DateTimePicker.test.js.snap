// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTimePicker should render properly 1`] = `
<PickerWithState
  KeyboardButtonProps={
    Object {
      "aria-label": "change date",
    }
  }
  allowKeyboardControl={true}
  ampm={true}
  autoOk="true"
  disableFuture={false}
  disablePast={false}
  format="dd/MM/yyyy HH:mm"
  id="date-picker-dialog"
  invalidDateMessage="Invalid Date Format"
  margin="normal"
  maxDate={2019-12-31T11:01:58.135Z}
  maxDateMessage="Date should not be after maximal date"
  minDate={2019-01-01T11:01:58.135Z}
  minDateMessage="Date should not be before minimal date"
  onChange={[MockFunction]}
  openTo="date"
  orientation="portrait"
  showTabs={true}
  value={2020-04-22T03:17:48.603Z}
  variant="inline"
  views={
    Array [
      "year",
      "date",
      "hours",
      "minutes",
    ]
  }
  wider={true}
/>
`;
